#51 세션 정보와 검색 정보를 담는 redis를 두개로 분리하여 저장 및 조회하도록 수정

0hun opened the pull request:
세션 정보를 담는 redis와 조회 캐시 정보를 담는 redis에 나눠서 저장하도록 수정
접속 정보(host, port) 수정

커넥션 빈 캐시 매니져를 redis에 맞게 각각 설정 및 추가
지역 정보 static 변수 추가

0hun committed:
#51 세션 정보와 검색 정보를 담는 redis를 두개로 분리하여 저장 및 조회하도록 수정

세션 정보를 담는 redis와 조회 캐시 정보를 담는 redis에 나눠서 저장하도록 수정
접속 정보(host, port) 수정

커넥션 빈 캐시 매니져를 redis에 맞게 각각 설정 및 추가
지역 정보 static 변수 추가

f-lab-dev commented on src/main/java/com/younghun/hibusgo/config/RedisConfig.java:
```diff
@@ -45,11 +53,17 @@
   }
 
+  @Bean
+  public RedisConnectionFactory connectionFactory2() {
```
이름에 숫자가 들어가면 유지보수하는 사람들이 이 bean의 사용처를 알 수 있을까요? 이름은 용도와 목적에 맞게 잘 지어주세요~

0hun replied:
ㅌㄴㅋ

0hun replied:
넵 수정하겠습니다.

f-lab-dev commented on src/main/java/com/younghun/hibusgo/config/RedisConfig.java:
```diff
@@ -45,11 +53,17 @@
    * - 비동기로 요청하기 때문에 Jedis에 비해 높은 성능을 가지고 있다.
    * - TPS, 자원사용량 모두 Jedis에 비해 우수한 성능을 보인다는 테스트 사례가 있다.
    */
+  @Primary
```
이렇게하면 bean의 이름을 지정하지 않을경우 세션레디스를 무조건 사용하게될텐데 그게 의도된 것일까요?
의도하지 않은 상황이 발생하면 에러를 발생시켜서 서버를 띄우지 못하게 하는 것도 필요합니다.

0hun replied:
bean의 이름을 지정하지 않으면, 메소드 이름으로 빈이 생성되지 않나요?. 같은 타입의 bean이 있을경우 @Primary가 붙은 메소드를 bean을 생성하고 bean의 이름 또는 메소드 호출시에 해당 bean을 호출하는걸로 알고있습니다. 실제로 테스트 해보니 캐시용 레디스 저장하는것을 확인하였습니다.

f-lab-dev replied:
네 메소드 이름으로 생성됩니다. 그리고 `@Primary`의 동작을 다시 봐보시면 좋을 것 같아요~ bean을 호출한다라기보단 주입한다라는 용어가 더 맞는 것 같고, 이런 bean의 우선순위를 지정하는 어노테이션을 쓸때는 의도되지 않은 상황을 만들 수 있는지 충분히 검토해봐야합니다

0hun replied:
넵넵. 찾아보니 RedisConnectionFactory의 경우 스프링 부트 자동 설정중에 만들어지는 빈들이라서 @Primary 사용을 해야된다고 합니다. rumtime시 타입을 판별하여 컴파일 문제를 잡기 위해 Type-safe한 @Primary를 권장한다고 합니다. 빈 주입단계에서는 빈의 이름을 통해 구분지어 준다면 빈의 이름을 찾아 해당 타입의 빈을 주입해주시때문에 문제되지 않을거라고 생각합니다.

https://www.inflearn.com/questions/5780

f-lab-dev commented on src/main/resources/application.properties:
```diff
@@ -18,9 +18,13 @@ mybatis.mapper-locations=classpath:mapper/*.xml
+spring.redis.timeout=360000
+
+#redis 설정(조회 정보 캐시용)
+spring.redis.cacheHost=localhost
```
`spring.redis.cache.host` 이런식으로 지어주는건 어떨까요?

0hun replied:
넵 수정하겠습니다.

0hun replied:
https://j0free.tistory.com/3 블로그 리뷰 부탁드립니다.

0hun committed:
#51 리뷰 수정

조회 정보 캐시용 접속 정보 변경
bean의 생성시 메소드 이름을 용도에 맞게 수정
bean 이름 지정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/config/RedisConfig.java:
```diff
@@ -54,13 +54,13 @@
   @Primary
-  @Bean
-  public RedisConnectionFactory connectionFactory() {
+  @Bean("connectionSessionRedisFactory")
```
안의 bean 이름은 생략해주셔도 될 것 같습니다~

0hun replied:
넵 수정하겠습니다.

f-lab-dev commented on src/main/java/com/younghun/hibusgo/config/RedisConfig.java:
```diff
@@ -54,13 +54,13 @@
-  @Bean
-  public RedisConnectionFactory connectionFactory() {
+  @Bean("connectionSessionRedisFactory")
+  public RedisConnectionFactory connectionSessionRedisFactory() {
```
클래스명이 `RedisConnectionFactory`니까 `sessionRedisConnectionFactory`가 좋지 않을까요?

0hun replied:
넵 수정하겠습니다.

0hun committed:
#51 리뷰 반영

사용 목적 및 의도가 명확하게 드러나도록 RedisConnectionFactory 이름 수정

0hun committed:
#51 리뷰 반영

redisCacheManager 빈 등록시 메소드 호출 방식이 아닌 RedisConnectionFactory의 변수 이름 과 해당 타입의
빈을 찾아 주입하고 해당 빈이 있어야 redisCacheManager 빈을 등록하도록 수정
